modules/db/src/main/scala/gem/dao/SmartGcalDao.scala

```diff
@@ -0,0 +1,24 @@
+package gem.dao
+
+import gem.config.{F2SmartGcalKey, GcalConfig, SmartGcalKey}
+import gem.enum.{GcalBaselineType, GcalLampType}
+import doobie.imports._
+
+object SmartGcalDao {
+  def insert(l: GcalLampType, b: GcalBaselineType, k: SmartGcalKey, g: GcalConfig): ConnectionIO[Int] = {
+    def insertSmartF2(gcalId: Int, k: F2SmartGcalKey): ConnectionIO[Int] =
+      sql"""
+        INSERT INTO smart_f2 (lamp, baseline, disperser, filter, fpu, gcal_id)
+             VALUES ($l :: gcal_lamp_type, $b :: gcal_baseline_type, ${k.disperser}, ${k.filter}, ${k.fpu}, $gcalId)
+      """.update.run
+
+    for {
+      id <- GcalDao.insert(g)
+      r  <-
+        k match {
```

Comment:
The key is a `SmartGcalKey`.  Right now, there's only the one instance, `F2SmartGcalKey`, but presumably there will be one for every instrument that can do smart gcal.  They are subsets of the full instrument config.  Only values that go into determining the calibration configuration are needed.  In the case of F2, that is the disperser, filter and FPU.

So, every instrument that supports smart gcal would need to be added to the match statement and an `INSERT` statement specific to its type will be needed.  Of course, if you create an instance of `SmartGcalKey` but fail to add it to the `match` statement the compiler will flag it as an error.